src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+public abstract class Figure implements HasArea, HasPerimeter {
+    private Color color;
+
+    public Figure(Color color) {
+        this.color = color;
+    }
+
+    public Color getColor() {
+        return color;
+    }
+
+    public void setColor(Color color) {
+        this.color = color;
+    }
+
+    public abstract int area();
+
+    public abstract int perimeter();
+
+    @Override
+    public String toString() {
+        return "Figure: " + this.getClass().getSimpleName() + ", area: " + area()
+                + ", perimeter: " + perimeter();
+    }
```

Comment:
according to task we should not override toString() method but create separate interface (e.g. Drawable) and make void method in it that will print information about figure.
```suggestion
    @Override
    public void draw() {
        System.out.printl("Figure: " + this.getClass().getSimpleName() + ", area: " + area()
                + ", perimeter: " + perimeter());
    }
```
Please also include information about color